src/Model/AuditReaderInterface.php

```diff
@@ -19,46 +19,47 @@
 interface AuditReaderInterface
 {
     /**
-     * @param string $className
-     * @param string $id
-     * @param string $revision
+     * @template T of object
      *
-     * @return object
+     * @param mixed $id
+     * @param mixed $revisionId
+     *
+     * @phpstan-param class-string<T> $className
+     * @phpstan-return T|null
      */
-    public function find($className, $id, $revision);
+    public function find(string $className, $id, $revisionId): ?object;
 
     /**
-     * @param string $className
-     * @param int    $limit
-     * @param int    $offset
-     *
      * @return object[]
+     *
+     * @phpstan-param class-string $className
      */
-    public function findRevisionHistory($className, $limit = 20, $offset = 0);
+    public function findRevisionHistory(string $className, int $limit = 20, int $offset = 0): array;
 
     /**
-     * @param string $classname
-     * @param string $revision
+     * @param mixed $revisionId
      *
-     * @return object
+     * @phpstan-param class-string $className
      */
-    public function findRevision($classname, $revision);
+    public function findRevision(string $className, $revisionId): ?object;
 
     /**
-     * @param string $className
-     * @param string $id
+     * @param mixed $id
      *
      * @return object[]
+     *
+     * @phpstan-param class-string $className
      */
-    public function findRevisions($className, $id);
+    public function findRevisions(string $className, $id): array;
 
     /**
-     * @param string $className
-     * @param int    $id
-     * @param int    $oldRevision
-     * @param int    $newRevision
+     * @param mixed $id
+     * @param mixed $oldRevisionId
+     * @param mixed $newRevisionId
+     *
+     * @return array<string, array{old: mixed, new: mixed, same: mixed}>
      *
-     * @return array
+     * @phpstan-param class-string $className
```

Comment:
The same here (native type hint is set). 
https://phpstan.org/writing-php-code/phpdocs-basics#combining-phpdoc-types-with-native-typehints

Reply:
This phpdoc makes phpstan more aware of what is this string, not a normal one, but a class string

Reply:
https://phpstan.org/writing-php-code/phpdoc-types#class-string :+1: Now it is clear for me.